bench/wtperf/wtperf.c

```diff
@@ -2957,14 +2889,35 @@ static uint64_t
 wtperf_rand(WTPERF_THREAD *thread)
 {
 	CONFIG_OPTS *opts;
+	WT_CURSOR *rnd_cursor;
 	WTPERF *wtperf;
 	double S1, S2, U;
 	uint64_t rval;
+	int ret;
+	char *key_buf;
 
 	wtperf = thread->wtperf;
 	opts = wtperf->opts;
 
 	/*
+	 * If we have a random cursor set up then use it.
+	 */
+	if ((rnd_cursor = thread->rand_cursor) != NULL) {
+		if ((ret = rnd_cursor->next(rnd_cursor))) {
+			lprintf(wtperf, ret, 0, "worker: rand next failed");
+			/* 0 is outside the expected range. */
+			return (0);
+		}
+		if ((ret = rnd_cursor->get_key(rnd_cursor, &key_buf)) != 0) {
+			lprintf(wtperf, ret, 0,
+			    "worker: rand next key retrieval");
+			return (0);
+		}
+		extract_key(key_buf, &rval);
```

Comment:
I think it's worth calling ``rand_cursor->reset`` here - I'd feel more comfortable both not keeping a positioned cursor and starting the random search from the beginning each time.